ee/modules/030-cloud-provider-openstack/templates/cloud-data-discoverer/deployment.yaml

```diff
@@ -0,0 +1,185 @@
+{{- define "openstack_cloud_data_discoverer_resources" }}
+cpu: 25m
+memory: 50Mi
+{{- end }}
+
+{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+---
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: cloud-data-discoverer
+  namespace: d8-cloud-provider-openstack
+  {{- include "helm_lib_module_labels" (list . (dict "app" "cloud-data-discoverer" "workload-resource-policy.deckhouse.io" "master")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: cloud-data-discoverer
+  updatePolicy:
+    updateMode: "Initial"
+  resourcePolicy:
+    containerPolicies:
+    - containerName: "cloud-data-discoverer"
+      minAllowed:
+        {{- include "openstack_cloud_data_discoverer_resources" . | nindent 8 }}
+      maxAllowed:
+        cpu: 50m
+        memory: 50Mi
+    {{- include "helm_lib_vpa_kube_rbac_proxy_resources" . | nindent 4 }}
+{{- end }}
+---
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: cloud-data-discoverer
+  namespace: d8-cloud-provider-openstack
+  {{- include "helm_lib_module_labels" (list . (dict "app" "cloud-data-discoverer")) | nindent 2 }}
+spec:
+  maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: cloud-data-discoverer
+---
```

Comment:
First, we can not use cluster autoscaler here because discoverer is located in the 030-cloud-provider-* module, cluster autoscaler in the 040-node-manager module.

Second, we can use discoverer for discovery cloud parameters for UI in the future.